internal/cloud/backend_taskStage_taskResults.go

```diff
@@ -67,20 +67,23 @@ func (trs *taskResultSummarizer) Summarize(context *IntegrationContext, output I
 func summarizeTaskResults(taskResults []*tfe.TaskResult) *taskResultSummary {
 	var pendingCount, errCount, errMandatoryCount, passedCount int
 	for _, task := range taskResults {
-		if task.Status == tfe.TaskUnreachable {
+		switch task.Status {
+		case tfe.TaskUnreachable:
 			return &taskResultSummary{
 				unreachable: true,
 			}
-		} else if task.Status == tfe.TaskRunning || task.Status == tfe.TaskPending {
+		case tfe.TaskPending, tfe.TaskRunning:
 			pendingCount++
-		} else if task.Status == tfe.TaskPassed {
+		case tfe.TaskPassed:
 			passedCount++
-		} else {
+		case tfe.TaskErrored, tfe.TaskFailed:
 			// Everything else is a failure
 			errCount++
 			if task.WorkspaceTaskEnforcementLevel == tfe.Mandatory {
 				errMandatoryCount++
 			}
+		default:
+			panic(fmt.Sprintf("unexpected tfe.TaskResultStatus: %#v", task.Status))
```

Comment:
I don't think that a PR for static checks should add a new panic call.
I know that the switch statement is handling all the task result types, but we should stick only to fixing linting issues.